Riot/Modules/profile/ProfileViewController.swift

```diff
@@ -0,0 +1,210 @@
+            return 1
+        case SectionType.PROFILE_CELL.rawValue:
+            return 1
+        case SectionType.ROLE_CELL.rawValue:
+            return roles.count
+        case SectionType.ICON_ITEM_CELL.rawValue:
+            return iconItems.count
+        case SectionType.ONLY_TEXT_CELL.rawValue:
+            return textItems.count
+        default:
+            return 0
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch indexPath.section {
+        case SectionType.PROFILE_CELL.rawValue:
+            guard let profileCell = self.tableView.dequeueReusableCell(withIdentifier: SectionType.PROFILE_CELL.cellIdentifier, for: indexPath) as? ProfileTableViewCell else { return UITableViewCell() }
+            profileCell.selectionStyle = .none
+            profileCell.setDelegate(delegate: self)
+            return profileCell
+        case SectionType.ROLE_CELL.rawValue:
+            guard let roleCell = self.tableView.dequeueReusableCell(withIdentifier: SectionType.ROLE_CELL.cellIdentifier, for: indexPath) as? ProfileRoleTableViewCell else { return UITableViewCell() }
+            roleCell.setValue(role: roles[indexPath.row])
+            roleCell.selectionStyle = .none
+            return roleCell
+        case SectionType.ICON_ITEM_CELL.rawValue:
+            guard let iconItemCell = self.tableView.dequeueReusableCell(withIdentifier: SectionType.ICON_ITEM_CELL.cellIdentifier, for: indexPath) as? DrawerItemTableViewCell else { return UITableViewCell() }
+            iconItemCell.setValue(iconItem: iconItems[indexPath.row])
+            iconItemCell.selectionStyle = .default
+            return iconItemCell
+        case SectionType.ONLY_TEXT_CELL.rawValue:
+            guard let textCell = self.tableView.dequeueReusableCell(withIdentifier: SectionType.ONLY_TEXT_CELL.cellIdentifier, for: indexPath) as? DrawerTextTableViewCell else { return UITableViewCell() }
+            textCell.setValue(textItem: textItems[indexPath.row])
+            textCell.selectionStyle = .default
+            return textCell
+        case SectionType.ROLE_DIVIDER_CELL.rawValue, SectionType.ICON_ITEM_DIVIDER_CELL.rawValue:
+            guard let dividerCell = self.tableView.dequeueReusableCell(withIdentifier: SectionType.ICON_ITEM_DIVIDER_CELL.cellIdentifier, for: indexPath) as? ProfileSectionDividerTableViewCell else { return UITableViewCell() }
+            dividerCell.selectionStyle = .none
+            return dividerCell
+        default:
+            return UITableViewCell()
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        switch indexPath.section {
+            //case SectionType.PROFILE_CELL.rawValue:
+            
+            //case SectionType.ROLE_CELL.rawValue:
+            
+        case SectionType.ICON_ITEM_CELL.rawValue:
+            if iconItems[indexPath.row].text == "Settings" {
```

Comment:
Is "Settings defined else where in the code, if so can this be centralised into a constant.  Does this handle internationalisation?

Reply:
Updated these references to use localised strings, and to use those strings for comparison in determining the button actions.